fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java

```diff
@@ -5356,6 +5365,35 @@ public void replayModifyTableProperty(short opCode, ModifyTablePropertyOperation
         }
     }
 
+    private void setExternalTableAutoAnalyze(Map<String, String> properties, ModifyTablePropertyOperationLog info) {
+        if (properties.size() != 1) {
+            LOG.warn("External table property should contain exactly 1 entry.");
+        }
+        if (!properties.containsKey(PropertyAnalyzer.PROPERTIES_AUTO_ANALYZE_POLICY)) {
+            LOG.warn("External table property should only contain auto_analyze_policy");
+        }
+        String value = properties.get(PropertyAnalyzer.PROPERTIES_AUTO_ANALYZE_POLICY);
+        if (!PropertyAnalyzer.ENABLE_AUTO_ANALYZE_POLICY.equalsIgnoreCase(value)
+                && !PropertyAnalyzer.DISABLE_AUTO_ANALYZE_POLICY.equalsIgnoreCase(value)
+                && !PropertyAnalyzer.USE_CATALOG_AUTO_ANALYZE_POLICY.equalsIgnoreCase(value)) {
+            LOG.warn("External table property should be 'enable', 'disable' or 'base_on_catalog'");
+            return;
+        }
+        try {
+            CatalogIf catalog = Env.getCurrentEnv().getCatalogMgr().getCatalog(info.getCtlName());
+            DatabaseIf db = catalog.getDbNullable(info.getDbName());
+            TableIf table = db.getTableNullable(info.getTableName());
+            if (!(table instanceof HMSExternalTable)) {
+                LOG.warn("Only support HMSExternalTable set property.");
+                return;
+            }
+            value = value.equalsIgnoreCase(PropertyAnalyzer.USE_CATALOG_AUTO_ANALYZE_POLICY) ? null : value;
+            ((HMSExternalTable) table).setAutoAnalyzePolicy(value);
```

Comment:
can just call catalog's method.
And I think we can not only support HMSTable, but all kind of catalogs.
You can do the check when doing the analyze.